File: finance-ui/src/api/files.ts
/**
 * File upload and preview API
 */
import apiClient from './client';

export interface FileUploadResponse {
  filename: string;
  path: string;
  size: number;
  sheets: string[];
}

export interface FileUploadListResponse {
  uploaded_files: FileUploadResponse[];
}

export interface FilePreviewResponse {
  filename: string;
  sheets: {
    name: string;
    headers: string[];
    rows: string[][];
    total_rows: number;
    total_columns: number;
  }[];
}

export const fileApi = {
  /**
   * Upload files
   */
  uploadFiles: async (files: File[]): Promise<FileUploadListResponse> => {
    const formData = new FormData();
    files.forEach((file) => {
      formData.append('files', file);
    });

    const response = await apiClient.post<FileUploadListResponse>(
      '/files/upload',
      formData,
      {
        headers: {
          'Content-Type': 'multipart/form-data',
        },
      }
    );
    return response.data;
  },

  /**
   * Get file preview
   */
  getFilePreview: async (filePath: string, maxRows: number = 100): Promise<FilePreviewResponse> => {
    const response = await apiClient.get<FilePreviewResponse>('/files/preview', {
      params: { file_path: filePath, max_rows: maxRows },
    });
    return response.data;
  },
};

File: finance-mcp/api/routers/files.py
"""
File upload and preview router
"""
from fastapi import APIRouter, Depends, UploadFile, File, HTTPException, Query
from typing import List
from api.models.user import User
from api.schemas.file import FileUploadListResponse, FileUploadResponse, FilePreviewResponse
from api.services.file_service import FileService
from api.routers.auth import get_current_user

router = APIRouter(prefix="/files", tags=["Files"])


@router.post("/upload", response_model=FileUploadListResponse)
async def upload_files(
    files: List[UploadFile] = File(..., description="Files to upload"),
    current_user: User = Depends(get_current_user)
):
    """
    Upload multiple files
    """
    if not files:
        raise HTTPException(status_code=400, detail="No files provided")

    uploaded_files = FileService.upload_multiple_files(files)

    return FileUploadListResponse(
        uploaded_files=[FileUploadResponse(**f) for f in uploaded_files]
    )


@router.get("/preview", response_model=FilePreviewResponse)
def preview_file(
    file_path: str = Query(..., description="File path to preview"),
    max_rows: int = Query(100, ge=1, le=1000, description="Maximum rows to preview"),
    current_user: User = Depends(get_current_user)
):
    """
    Get Excel file preview data
    """
    preview_data = FileService.get_file_preview(file_path, max_rows)
    return FilePreviewResponse(**preview_data)
